oeps/oep-0026-arch-xapi-integration.rst

```diff
@@ -0,0 +1,483 @@
+(PII)`_, like the learner's email address or name, then those may be conditionally sent
+with appropriate permissions. Adaptive engines, however, do not need PII.
+
+Initially, we will exclude the "name" field. However, if we find that xAPI JSON parsers
+assume this field always exists, then we can include the field but provide a non-PII
+value, such as a copy of the `Anonymized User ID`_.
+
+Anonymized User ID
+^^^^^^^^^^^^^^^^^^
+An *Anonymized User ID* will be generated as a SHA-256 hash of the Open edX user_id, 
+which is the **id** value of the user's row in the `Django auth_user`_ table in the
+Open edX LMS.
+
+As mentioned in `Using consistent user identifiers in Segment events`_, the other user
+identifiers in the Open edX platform aren't perfectly suited for inclusion in externally
+sent events:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Identifier
+     - Limitations
+   * - username
+     - Often contains PII, such as user's name or year of birth.
+   * - LMS user_id
+     - Information leakage of auth_user table with easily guessable values; tied down
+       to implementation of the housing database.
+   * - email address
+     - Is PII and modifiable by the user.
+   * - anonmyous user_id
+     - It is currently constructed by hasing the user's LMS user_id with the Django
+       server's *SECRET_KEY* value. This value will change when the *SECRET_KEY* is rotated.
+
+.. _Django auth_user: https://docs.djangoproject.com/en/2.0/topics/auth/default/#user-objects
+
+Example
+^^^^^^^
+
+Here is an example of an **Actor** JSON value that we would generate:
+
+::
+
+    "actor": {
+        “objectType”: “Agent”,
+        “openid”: “https://users.openedx.org/1234567890”,
+        "name": “https://users.openedx.org/1234567890”  # only include this field if necessary
+    }
+
+See `Deep Dive: Actor/Agent`_ for more information on xAPI Actors.
+
+.. _Friend of a Friend (FOAF): http://xmlns.com/foaf/spec/
+.. _email address: http://xmlns.com/foaf/spec/#term_mbox
+.. _hash of email address: http://xmlns.com/foaf/spec/#term_mbox_sha1sum
+.. _OpenID URI: http://xmlns.com/foaf/spec/#term_openid
+.. _account: http://xmlns.com/foaf/spec/#term_account
+.. _Personally Identifiable Information (PII): https://en.wikipedia.org/wiki/Personally_identifiable_information
+.. _Using consistent user identifiers in Segment events: https://openedx.atlassian.net/wiki/spaces/AN/pages/144441849/Using+consistent+user+identifiers+in+Segment+events
+.. _`Deep Dive: Actor/Agent`: https://xapi.com/deep-dive-actor-agent/
+
+xAPI Verb
+~~~~~~~~~
+
+The **Verb** in xAPI is a past tensed value, identified by a URI from the `xAPI registry`_
+and paired with a short display string. It denotes the action that was performed by the
+Actor on the Object in the statement. We will use only values found in the registry and
+document which values are used for which edX events in the `Open edX Events`_ section below.
```

Comment:
Sure.  I mean to say that we will try to use standard and registered Verbs where ever possible - rather than creating our own custom ones.  I'll update the text to make this clearer.